lib/compat/wordpress-6.0/class-wp-webfonts.php

```diff
@@ -81,19 +107,45 @@ public function get_providers() {
 	public function register_font( $font ) {
 		$font = $this->validate_font( $font );
 		if ( $font ) {
-			$id                    = $this->get_font_id( $font );
-			$this->webfonts[ $id ] = $font;
+			$slug = $this->get_font_slug( $font );
+
+			if ( ! isset( $this->registered_webfonts[ $slug ] ) ) {
+				$this->registered_webfonts[ $slug ] = array();
+			}
+
+			$this->registered_webfonts[ $slug ][] = $font;
+		}
+	}
+
+	public function enqueue_font( $font_family ) {
+		$slug = $this->get_font_slug( $font_family );
+
+		if ( isset( $this->enqueued_webfonts[ $slug ] ) ) {
+			_doing_it_wrong( __FUNCTION__, sprintf( __( 'The "%s" font family is already enqueued.' ), $slug ), '6.0.0' );
```

Comment:
Each case of `__doing_wrong()` is being reviewed and more often than not replaced by `trigger_error()`. Why? Per @azaozz, `_doing_it_wrong()` is "yelling" at the developer and it should be reserved for the most egregious instances.

What about here? Does this qualify as an egregious instance? I don't think "already enqueued" rises to the level of yelling at the developer. Instead, I think using `trigger_error()` is more appropriate as it will log the issue to alert the developer that action is needed.

```suggestion
			trigger_error( 
				sprintf(
					/* translators: %s unique slug to identify the webfont */
					__( 'The "%s" font family is already enqueued.', 'gutenberg' ),
					$slug
				)
			);
```